### PR TITLE
build: Update Strum to Latest Version

### PR DESCRIPTION
**Pull request by evolvedmicrobe**
Currently is < 0.24, but figure we'll need to cross that bridge eventually.

### CODE REVIEW

**Comment by evolvedmicrobe** [comment]
See companion PR at: https://github.com/rust-bio/rust-bio/pull/491

File: Cargo.toml [hunk 1]
```diff
@@ -23,4 +23,4 @@ regex = "1.0"
 lazy_static = "1.1"
 derive-new = "0.5"
 petgraph = { version = ">=0.5,<0.7", optional = true }
-strum_macros = ">=0.20, <0.24"
+strum_macros = "0.24"
```
Comment (adam-azarchs) [inline Cargo.toml]: I see no reason not to keep this as a version range here, to remove the need for everyone to update everything in sync.  Nothing here is taking advantage of any features which are only present in the latest version.
```suggestion
strum_macros = ">=0.20, <0.25"
```

**Comment by adam-azarchs** [comment]
If you want to actually be able to release a new version after this merges, you should update the PR title to `build:` rather than `chore:`, since the release bot ignores chores.

**Review by pmarks — COMMENTED** [review]
lgtm

**Review by pmarks — APPROVED** [review]

**Pull request merged**